src/ophys_etl/modules/segmentation/modules/calculate_edges.py

```diff
@@ -0,0 +1,125 @@
+            nx.read_gpickle(graph_path),
+            video_path)
+    nx.write_gpickle(graph, graph_path)
+    return graph_path
+
+
+def filtered_pearson_edge_job(filter_fraction: float,
+                              graph_path: Path,
+                              video_path: Path) -> nx.Graph:
+    graph = edge_attributes.add_filtered_pearson_edge_attributes(
+            filter_fraction,
+            nx.read_gpickle(graph_path),
+            video_path)
+    nx.write_gpickle(graph, graph_path)
+    return graph_path
+
+
+def hnc_gaussian_edge_job(
+        neighborhood_radius: int,
+        graph_path: Path,
+        video_path: Path,
+        attribute_name: str = 'hnc_Gaussian',
+        filter_fraction: Optional[float] = None,
+        full_neighborhood: bool = False) -> nx.Graph:
+    graph = edge_attributes.add_hnc_gaussian_metric(
+            graph=nx.read_gpickle(graph_path),
+            video_path=video_path,
+            neighborhood_radius=neighborhood_radius,
+            filter_fraction=filter_fraction,
+            attribute_name=attribute_name,
+            full_neighborhood=full_neighborhood)
+    nx.write_gpickle(graph, graph_path)
+    return graph_path
+
+
+class CalculateEdges(argschema.ArgSchemaParser):
+    default_schema = CalculateEdgesInputSchema
+
+    def run(self):
+        self.logger.name = type(self).__name__
+        t0 = time.time()
+
+        # select which calculation to perform
+        if self.args['attribute_name'] == 'Pearson':
+            edge_job = pearson_edge_job
+        elif self.args['attribute_name'] == 'filtered_Pearson':
+            edge_job = partial(filtered_pearson_edge_job,
+                               self.args['filter_fraction'])
+        elif self.args['attribute_name'] == 'hnc_Gaussian':
+            edge_job = partial(hnc_gaussian_edge_job,
+                               self.args['neighborhood_radius'],
+                               attribute_name=self.args['attribute_name'])
+        elif self.args['attribute_name'] == 'filtered_hnc_Gaussian':
+            edge_job = partial(
+                          hnc_gaussian_edge_job,
+                          self.args['neighborhood_radius'],
+                          filter_fraction=self.args['filter_fraction'],
+                          attribute_name=self.args['attribute_name'],
+                          full_neighborhood=self.args['full_neighborhood'])
+
+        if "graph_input" not in self.args:
+            with h5py.File(self.args["video_path"], "r") as f:
+                shape = f["data"].shape[1:]
+            graph = creation.create_graph(0, shape[0] - 1, 0, shape[1] - 1,
+                                          self.args["kernel"])
+            nx.write_gpickle(graph, self.args["graph_output"])
+            self.args["graph_input"] = self.args["graph_output"]
+            self.logger.name = type(self).__name__
+
+        if self.args["n_parallel_workers"] == 1:
```

Comment:
Is this branch necessary? Wouldn't `partition.partition_graph_by_edges` where `"n_parallel_workers" = 1 be the same?

Reply:
I like to have a branch where multiprocessing is not invoked at all, even with one job. It can help measure test coverage, and, I find it helpful in debugging to eliminate multiprocessing when looking at a stack trace.